### DISCUSSION TITLE
Possible to get list of loaded mods?

### DISCUSSION THREAD

**Discussion by willuwontu** [BepInEx/BepInEx#320] [Help and Q&A] [answered]
Hey guys,

Is it possible to get a list of mods that bepinex has loaded in?

**Comment by ManlyMarco** [comment, marked as answer]
Read the game log or the bepinex log, or enable bepinex console.

**Reply by willuwontu** [reply]
Let me clarify, I want to get the list of loaded mods from within my mod, so that I can execute different code based upon the presence of another mod.

**Reply by ghorsington** [reply]
In that case, you can use [`Chainloader.PluginInfos`](https://docs.bepinex.dev/api/BepInEx.Bootstrap.Chainloader.html#BepInEx_Bootstrap_Chainloader_PluginInfos) dictionary to check if a plugin was loaded. The dictionary key is the GUID of the plugin.

If you do use this, you might also want to set that plugin as a soft dependency with [`BepInDependency`](https://docs.bepinex.dev/api/BepInEx.BepInDependency.html#BepInEx_BepInDependency__ctor_System_String_BepInEx_BepInDependency_DependencyFlags_) attribute. Setting the plugin as the soft dependency ensures your plugin is loaded after the dependency is loaded so that the `PluginInfos` dictionary has the other plugin if it was installed by the user.

**Reply by willuwontu** [reply]
That's exactly what I needed, thanks!

**Reply by xiaoxiao921** [reply]
I got outspeeded, in any case here is the code version of what ghorsington just described

```cs
[BepInDependency(modGuid, BepInDependency.DependencyFlags.SoftDependency)]
class MyMod : BaseUnityPlugin
{
	void Awake()
	{
		// Only already loaded mods show up in PluginInfos, thats why SoftDependency is used above to ensure that.
		foreach (var plugin in Chainloader.PluginInfos)
		{
			var metadata = plugin.Value.Metadata;
			if (metadata.GUID.Equals(modGuid))
			{
				// found it
				Logger.Info($"Found {modGuid}");
				break;
			}
		}
	}
}
```
